### ISSUE TITLE
5.3.3 对文本生成函数进行调整generate代码的缩紧有误

### ISSUE THREAD

**Issue by bhxyry** [skindhu/Build-A-Large-Language-Model-CN#29]
if temperature > 0.0:                                       #C
        logits = logits / temperature
        probs = torch.softmax(logits, dim=-1)
        idx_next = torch.multinomial(probs, num_samples=1)
    else:                                                       #D
        idx_next = torch.argmax(logits, dim=-1, keepdim=True)
    if idx_next == eos_id:                                      #E
        break
    idx_next = idx_next.unsqueeze(1)
    idx = torch.cat((idx, idx_next), dim=1)
return idx
以后缩进少一个tab
